bundler/lib/bundler/compact_index_client/updater.rb

```diff
@@ -1,46 +1,60 @@
 # frozen_string_literal: true
 
 require_relative "../vendored_fileutils"
+require "rubygems/package"
 
 module Bundler
   class CompactIndexClient
     class Updater
+      # TODO: support more algorithms, at least sha-512
+      SUPPORTED_DIGESTS = { "sha-256" => :SHA256 }.freeze
+      # The value is relative preferences from 1=least to 10=most.
+      WANT_DIGEST = SUPPORTED_DIGESTS.keys.map.with_index {|a, i| "#{a}=#{10-i}" }.join(", ").freeze
+
       class MisMatchedChecksumError < Error
-        def initialize(path, server_checksum, local_checksum)
+        def initialize(path, server_checksums, local_checksums)
           @path = path
-          @server_checksum = server_checksum
-          @local_checksum = local_checksum
+          @server_checksums = server_checksums
+          @local_checksums = local_checksums
         end
 
         def message
           "The checksum of /#{@path} does not match the checksum provided by the server! Something is wrong " \
-            "(local checksum is #{@local_checksum.inspect}, was expecting #{@server_checksum.inspect})."
+            "(local checksum is #{@local_checksums.inspect}, was expecting #{@server_checksums.inspect})."
         end
       end
 
       def initialize(fetcher)
         @fetcher = fetcher
       end
 
-      def update(local_path, remote_path, retrying = nil)
+      def update(local_path, remote_path, local_etag_path, retrying = nil)
         headers = {}
 
         local_temp_path = local_path.sub(/$/, ".#{$$}")
         local_temp_path = local_temp_path.sub(/$/, ".retrying") if retrying
+        local_etag_temp_path = local_temp_path.sub(/$/, ".etag.tmp")
+        local_etag_temp_path = local_temp_path.sub(/$/, ".retrying") if retrying
         local_temp_path = local_temp_path.sub(/$/, ".tmp")
 
+        headers["Want-Content-Digest"] = WANT_DIGEST
```

Comment:
I'm not seeing `Want-Content-Digest` listed at all on MDN, and it says `Want-Digest` is deprecated https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Want-Digest

Reply:
It's only a [draft](https://datatracker.ietf.org/doc/html/draft-ietf-httpbis-digest-headers) 🤷‍♂️

Reply:
We could totally skip this header. It's just future proofing.